src/main/utils/get-port.ts

```diff
@@ -0,0 +1,51 @@
+import { Readable } from "stream";
+import URLParse from "url-parse";
+
+interface GetPortArgs {
+  /**
+   * Should be case insensitive
+   * Must have a named matching group called `address`
+   */
+  lineRegex: RegExp;
+  /**
+   * Called when the port is found
+   */
+  onFind?: () => void;
+  /**
+   * Timeout for how long to wait for the port.
+   * Default: 5s
+   */
+  timeout?: number;
+}
+
+/**
+ * Parse lines from `stream` (assumes data comes in lines) to find the port
+ * which the source of the stream is watching on.
+ * @param stream A readable stream to match lines against
+ * @param args The args concerning the stream
+ * @returns A Promise for port number
+ */
+export function getPortFrom(stream: Readable, args: GetPortArgs): Promise<number> {
```

Comment:
Would make sense to write tests for this.

Reply:
Very much so